SSA/Projects/InstCombine/LLVM/PrettyEDSL.lean

```diff
@@ -47,13 +47,12 @@ macro_rules
 
 open MLIR.AST
 
-syntax mlir_op_operand " = " "llvm.mlir.constant" neg_num (" : " mlir_type)? : mlir_op
-syntax mlir_op_operand " = " "llvm.mlir.constant" ("$" noWs "{" term "}") (" : " mlir_type)? : mlir_op
+syntax mlir_op_operand " = " "llvm.mlir.constant(" neg_num (" : " mlir_type)? ")" (" : " mlir_type) : mlir_op
+syntax mlir_op_operand " = " "llvm.mlir.constant(" ("$" noWs "{" term "}") ")" (" : " mlir_type)?   : mlir_op
 macro_rules
-  | `(mlir_op| $res:mlir_op_operand = llvm.mlir.constant $x $[: $t]?) => do
-      let t ← t.getDM `(mlir_type| _)
-      `(mlir_op| $res:mlir_op_operand = "llvm.mlir.constant"() {value = $x:neg_num : $t} : () -> ($t) )
-  | `(mlir_op| $res:mlir_op_operand = llvm.mlir.constant ${ $x:term } $[: $t]?) => do
+  | `(mlir_op| $res:mlir_op_operand = llvm.mlir.constant( $x $[: $t]?) : $typ ) => do
+      `(mlir_op| $res:mlir_op_operand = "llvm.mlir.constant"() {value = $x:neg_num : $typ} : () -> ($typ) )
```

Comment:
This implementation will completely ignore the internal type annotation `$t`.
We should check that if `t` is given, it's the same as `typ`.
And if `t` is not given, we should check that `typ` is either `f64` or `u64`, to be completely compliant with the LLVM docs